great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -2717,21 +2728,31 @@ def _initialize_usage_statistics(
 
     def _init_datasources(self) -> None:
         """Initialize the datasources in store"""
-        config: DataContextConfig = self.get_config_with_variables_substituted(
-            self.config
-        )
+        config: DataContextConfig = self.config
         datasources: Dict[str, DatasourceConfig] = cast(
             Dict[str, DatasourceConfig], config.datasources
         )
 
+        substitutions = self._determine_substitutions()
+
         for datasource_name, datasource_config in datasources.items():
             try:
                 config = copy.deepcopy(datasource_config)  # type: ignore[assignment]
```

Comment:
Why are we ignoring this?

Reply:
I'm not sure haha let's see if `mypy` complains